zict/tests/test_lru.py

```diff
@@ -88,6 +91,25 @@ def cb(k, v):
     assert count[0] == len(L)
 
 
+def test_cb_exception_keep_on_lru():
+    def cb(k, v):
+        raise Exception
+
+    L = list()
+    d = dict()
+    lru = LRU(2, d, on_evict=[lambda k, v: L.append((k, v)), cb])
+
+    lru["x"] = 1
+    lru["y"] = 2
+
+    with pytest.raises(Exception):
+        lru["z"] = 3
+
+    assert L == [("x", 1)]  # tried to evict and raised exception
+    assert lru.total_weight == 3
+    assert lru.weights == {"x": 1, "y": 1, "z": 1}
+
+
```

Comment:
+ Please test use case where individual weight >= n
+ Missing tests for Buffer

Reply:
Thanks for pointing this out, I'll be adding buffer tests soon.